src/accelerator.rs

```diff
@@ -0,0 +1,98 @@
+use drivers::Driver;
+use Motion;
+use Operation;
+
+pub struct Accelerator<D: Driver> {
+    motions: Vec<Motion>,
+    driver: D,
+}
+
+impl <D: Driver> Accelerator<D> {
+    pub fn new(mots: Vec<Motion>, drv: D) -> Self {
+        Accelerator{motions: mots, driver: drv}
+    }
+    // Needs to be un pub.ed
+    pub fn within_bounds(&self, n: isize) -> isize{
+        println!("called bounds with \'{}\'", n);
+        let min = 0;
+        let max = self.motions.len() as isize;
+        if n < min { return min }
+        if n > max { return max }
+        n
+    }
+
+    fn execute(&mut self, mut start: isize, mut finish: isize){
+        println!("called execute with \'{0}\' and \'{1}\'", start,finish);
+        if start != finish {
```

Comment:
I generally like the pattern of `if (start === finish) { return }` as it reduces cyclomatic complexity. Is there a reason to do it this way?